builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -377,6 +378,21 @@ func resourceVSphereVirtualMachine() *schema.Resource {
 							Default:  false,
 							ForceNew: true,
 						},
+
+						"controller_type": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "scsi",
+							ForceNew: true,
+							ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+								value := v.(string)
+								if value != "scsi" && value != "ide" {
+									errors = append(errors, fmt.Errorf(
```

Comment:
We also can check this in the schema with a validator ... Which we should be doing more.  I will help us long term to keep the variable error handling in the schema.